Documentation

UnseenFaith opened the pull request:
### Proposed Semver Increment Bump: [MAJOR/MINOR/PATCH]
Minor
### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Changing the structure of Komada to reflect Klasa's structure. This will make building documentation and linting easier.
- Adds Documentation to Komada


Almost all of the documentation is done besides:
- A few more classes (I think it's only settings stuff)
- Only one of the templates (commands) is made. I have yet to make the rest.

UnseenFaith committed:
Documentation

bdistin committed:
Update package.json

bdistin committed:
Create .travis-deploy.sh

bdistin committed:
Update .travis.yml

bdistin committed:
Create .createFolder

CyberiumShadow committed:
Fixes Merge Conflicts...

CyberiumShadow committed:
Fixes Merge Conflicts

CyberiumShadow committed:
Merge branch 'UnseenFaith-documentation' into documentation

CyberiumShadow committed:
Fixes Missing files

bdistin committed:
Rename docstrap.json to .docstrap.json

CyberiumShadow committed:
ESlint + Untracked Class

kyranet commented on src/classes/argResolver.js:
```diff
@@ -165,11 +317,21 @@ module.exports = class ArgResolver extends Resolver {
-};
+}
+
+module.exporots = ArgResolver;
```
lol

kyranet commented on src/classes/client.js:
```diff
@@ -69,21 +181,56 @@ module.exports = class Komada extends Discord.Client {
+   * @readonly
+   * @type {SchemaManager}
+   */
+  get schemaManager() {
```
This got removed in #313

kyranet reviewed (dismissed):
Well, besides that this PR reversed #313 and #322, it looks good to me.

kyranet committed:
Merged PRs 313 and 322

kyranet committed:
Added TextChannel and VoiceChannel to schemaManager

UnseenFaith committed:
Merge pull request #2 from dirigeants/merge-prs

Merged PRs 313 and 322

kyranet committed:
Fixed getResolved and other mini-bugs, added documentation.

Now getResolved should work perfectly when the SG instance does not aim for Guilds.

kyranet committed:
Added Collection and VoiceChannel

kyranet committed:
Deleted test

kyranet committed:
Added examples, fixed returns, defaulted forces to true.

kyranet committed:
Added names, documented better the constructor

kyranet committed:
Update settingResolver.js

kyranet committed:
Documented Komada's config with deep detail.

kyranet committed:
Update client.js

kyranet committed:
Documented settingsCache

kyranet committed:
Documenting, fixed multiple bugs

CacheManager is not longer for guild settings, and it was *still* hardcoded to use guilds instead of this.type

kyranet committed:
Klasa -> Komada

kyranet committed:
Fixed padding in comments.

kyranet committed:
Fixed paddings (thanks Dada) and added memberof

kyranet committed:
Better performance, made this.provider a getter...

This makes sense to me now.

kyranet committed:
Now the SQL class gets provider from Gateway.provider

UnseenFaith committed:
More documentation

UnseenFaith committed:
Fix copypasta

UnseenFaith committed:
Fix Global typedef && add examples/tutorials

UnseenFaith committed:
Fix docstrap and add example folder

kyranet committed:
Bugfix in SQL class

kyranet committed:
Fix typo #324

CyberiumShadow committed:
Merge branch 'documentation' of git://github.com/UnseenFaith/komada into UnseenFaith-documentation

CyberiumShadow committed:
Merge branch 'UnseenFaith-documentation' into documentation

CyberiumShadow committed:
Merge branch 'indev' into documentation

UnseenFaith committed:
Remove JSON Provider documentation

kyranet committed:
Merge branch 'indev' into documentation

kyranet committed:
Create ProviderStructure.js

kyranet committed:
Update clean.js

kyranet committed:
Fixed a serious bug.

kyranet committed:
Better schema parsing

kyranet committed:
Fixed the default in the doc.

kyranet committed:
Bumped version to 0.20.7

kyranet committed:
Update CHANGELOG.md

UnseenFaith committed:
Remove this testing thing.

kyranet reviewed (approved):
LGTM
